doc/nrf/protocols/matter/getting_started/adding_clusters.rst

```diff
@@ -94,17 +94,90 @@ This tool is provided with the Matter repository in the |NCS|.
+
+         Your currenly installed ZAP tool version: 2022.12.20 matches the recommended one.
+
+      It means that your ZAP version is correct and the tool executable can be accessed from the operating system environment, so you can skip the following steps regarding ZAP download.
+
+      * If your currently install ZAP version does not match the recommended one or no ZAP version is installed on your device, you will see the following messages:
+
+      .. code-block::
+
+         Your currenly installed ZAP tool version 2022.12.19 does not match the recommended one: 2022.12.20.
+
+      or
+
+      .. code-block::
+
+         No ZAP tool version was found installed on this device.
+
+      The package download process will start automatically afterwards:
+
+      .. code-block::
+
+         Trying to download ZAP tool package matching your system and recommended version.
+         100% [......................................................................] 150136551 / 150136551
+         ZAP tool package was downloaded and extracted in the given location.
+
+         #####################################################################################
+         # Please add the following location to the system PATH: package_extraction_location #
+         #####################################################################################
+
+      .. note::
+          The ``package_extraction_location`` in the example output will be replaced, in the real logs from the script with the extracted package location path on your device.
+
+   #. Add the ``package_extraction_location`` ZAP package location printed in the script output to the system :envvar:`PATH` environment variables.
+
+   .. tabs::
+
+      .. group-tab:: Linux
+
+         For example, if you are using ``bash``, run the following commands:
+
+         .. parsed-literal::
+            :class: highlight
+
+            echo 'export PATH=package_extraction_location:"$PATH"' >> ${HOME}/.bashrc
+            source ${HOME}/.bashrc
+
+      .. group-tab:: Windows
+
+         For the exact instructions, see :ref:`zephyr:env_vars`.
+
+      .. group-tab:: macOS
+
+         For example, if you are using ``bash``, run the following commands:
+
+         .. parsed-literal::
+            :class: highlight
+
+            echo 'export PATH=package_extraction_location:"$PATH"' >> ${HOME}/.bash_profile
+            source ${HOME}/.bash_profile
+   ..
+
 #. Open the :file:`src/template.zap` for editing by running the following command, where ``samples/matter/sensor`` stands for the path where you copied the template sample in the first step of this guide.
 
    .. code-block::
 
-      ./scripts/tools/zap/run_zaptool.sh ../../../nrf/samples/matter/sensor/src/template.zap
-
-   .. note::
-       The :file:`run_zaptool.sh` script will run the bootstrap procedure to install all required ZAP tool dependencies (Node.js packages).
+      zap ../../../nrf/samples/matter/sensor/src/template.zap
```

Comment:
Under windows the below line does not work (sorry have to comment here, i can't comment unmodified lines):

      `./scripts/tools/zap/generate.py ../../../nrf/samples/matter/sensor/src/template.zap`

This sort of works.. (there is different error).

      `python scripts/tools/zap/generate.py ../../../nrf/samples/matter/sensor/src/template.zap`

New error:

```
Traceback (most recent call last):
  File "scripts/tools/zap/generate.py", line 295, in <module>
    main()
  File "scripts/tools/zap/generate.py", line 278, in main
    runGeneration(cmdLineArgs.zapFile, cmdLineArgs.zclFile, cmdLineArgs.templateFile, cmdLineArgs.outputDir)
  File "scripts/tools/zap/generate.py", line 190, in runGeneration
    extractGeneratedIdl(output_dir, zap_file)
  File "scripts/tools/zap/generate.py", line 153, in extractGeneratedIdl
    os.rename(idl_path, target_path)
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\ncs\\v2.2.99-dev3\\modules\\lib\\matter\\..\\..\\..\\nrf\\samples\\matter\\template\\src\\zap-generated/Clusters.matter' -> 'C:\\ncs\\v2.2.99-dev3\\modules\\lib\\matter\\../../../nrf/samples/matter/template/src/template.matter'
```

Reply:
Regarding the first failure, ok, I didn't go through the unmodified part of guide, but it seems to be reasonable to add `python` at the beginning of the command. The second failure might be a problem, as it's an internal behavior of generate.py script that we do not develop. Maybe it requires adding some additional permissions on Windows. Probably nobody from upstream tested it with Windows.

Reply:
Issue with generate.py was fixed by the patch on corresponding sdk-connectedhomeip PR.